pkg/storage/system_intake_grb_reviewer.go

```diff
@@ -43,23 +43,51 @@ func (s *Store) CreateSystemIntakeGRBReviewers(ctx context.Context, np sqlutils.
 	return reviewers, nil
 }
 
-func (s *Store) UpdateSystemIntakeGRBReviewer(ctx context.Context, tx *sqlx.Tx, reviewerID uuid.UUID, votingRole models.SystemIntakeGRBReviewerVotingRole, grbRole models.SystemIntakeGRBReviewerRole) (*models.SystemIntakeGRBReviewer, error) {
+func (s *Store) UpdateSystemIntakeGRBReviewer(ctx context.Context, tx *sqlx.Tx, input *models.UpdateSystemIntakeGRBReviewerInput) (*models.SystemIntakeGRBReviewer, error) {
 	updatedReviewer := &models.SystemIntakeGRBReviewer{}
 	if err := namedGet(ctx, tx, updatedReviewer, sqlqueries.SystemIntakeGRBReviewer.Update, args{
-		"reviewer_id": reviewerID,
-		"grb_role":    grbRole,
-		"voting_role": votingRole,
+		"reviewer_id": input.ReviewerID,
+		"grb_role":    input.GrbRole,
+		"voting_role": input.VotingRole,
 		"modified_by": appcontext.Principal(ctx).Account().ID,
 	}); err != nil {
 		appcontext.ZLogger(ctx).Error(
 			"error updating system intake GRB reviewer",
-			zap.String("reviewer_id", reviewerID.String()),
+			zap.Error(err),
+			zap.String("reviewer_id", input.ReviewerID.String()),
 		)
+
+		return nil, err
 	}
 
 	return updatedReviewer, nil
 }
 
+// CastSystemIntakeGRBReviewerVote inserts or updates the vote selection for a GRB Reviewer
+// requires the requesting user to be the GRB reviewer casting the vote (i.e., an admin cannot set another user's vote)
+func (s *Store) CastSystemIntakeGRBReviewerVote(ctx context.Context, input models.CastSystemIntakeGRBReviewerVoteInput) (*models.SystemIntakeGRBReviewer, error) {
```

Comment:
Non-blocking, but a question for this. Would it make sense just to use update? This is basically just a selective update happening. We might simplify store logic if we just use the one store method but have distinct resolver methods in place. This shouldn't block merging this PR.